src/iotEdgeExplorer.ts

```diff
@@ -116,6 +116,35 @@ export class IoTEdgeExplorer extends BaseExplorer {
         return true;
     }
 
+    private async isValidCreateOptions(desiredProperties: any): Promise<boolean> {
+
+        try {
+            const moduleCreateOptions: string [] = [];
+
+            const systemModules = desiredProperties.systemModules;
+            moduleCreateOptions.push(systemModules.edgeAgent.settings.createOptions);
+            moduleCreateOptions.push(systemModules.edgeHub.settings.createOptions);
+
+            const customModules = desiredProperties.modules;
+            for (const moduleName in customModules) {
+                if (customModules.hasOwnProperty(moduleName)) {
+                    moduleCreateOptions.push(customModules[moduleName].settings.createOptions);
+                }
+            }
+
+            moduleCreateOptions.forEach( (createOption) => {
+                if (createOption) {
+                    JSON.parse(createOption);
+                }
+            });
+
+            return true;
+        } catch (error) {
+            vscode.window.showErrorMessage(`Cannot parse createOptions of modules: ${error.message}`);

```

Comment:
The message should be more clearer to user. And let user know which module has invalid createOptions.

Reply:
Improved